src/pages/ProfilePage.js

```diff
@@ -289,5 +299,20 @@ export default compose(
         betas: {
             key: ONYXKEYS.BETAS,
         },
+        session: {
+            key: ONYXKEYS.SESSION,
+        },
+    }),
+    // eslint-disable-next-line rulesdir/no-multiple-onyx-in-file
```

Comment:
I'm having second thoughts about this. What was the reason for adding the second `withOnyx` call? And could we fix this by just adding a `privateNotesReportID` to the user's session key maybe?

Reply:
> I'm having second thoughts about this. What was the reason for adding the second withOnyx call?

Because we are dependent on the value of `session` props. We have a few use cases with similar pattern.


> And could we fix this by just adding a privateNotesReportID to the user's session key maybe?

Users can have multiple private notes. On each chat, they have a different report. To say, private notes are report rNVPs characterized by the user's accountID(so each participant) can have their notes on chats they belong to. So do you think we can still add the key to the user's session?

Reply:
Hmm okay it sounds like these private notes reportIDs should probably go in it's own key/collection. Something like `privateNotesReportID_{accountID}` though we should probably take that to #engi-chat to get second opinions.

In the meantime, did we really need to change the way we load this to begin with? What was wrong with the logic we had before [here](https://github.com/Expensify/App/blame/083c924eb8d8fc2322d2b65d71d8b1ccaf896453/src/pages/ProfilePage.js#L139)?

Reply:
> Hmm okay it sounds like these private notes reportIDs should probably go in it's own key/collection. Something like privateNotesReportID_{accountID} though we should probably take that to #engi-chat to get second opinions.

Yeah, I did bring that up in the design doc but we went with adding it as key to report object.

Reply:
> In the meantime, did we really need to change the way we load this to begin with? What was wrong with the logic we had before [here](https://github.com/Expensify/App/blame/083c924eb8d8fc2322d2b65d71d8b1ccaf896453/src/pages/ProfilePage.js#L139)?

I seriously can't recall why I went with this change, I have reverted back everything to get rid of double `withOnyx` usage.

Reply:
> Yeah, I did bring that up in the design doc but we went with adding it as a key to report object. See here: https://docs.google.com/document/d/1StLBx3Y6yJsZp54rHMu-mGo20pUt1niV6Y0MPNDR6A4/edit?disco=AAAA0B4ClCw

If I'm understanding correctly, It sounds like these private note reports are tied to _accounts_ not reports right? If so I think the concerns in that thread were misguided 😄. Ideally we shouldn't have to load a report just to get the privateNotesReportID for an account. Either way, I guess there's no problem with the revert so we can probably just :donothing:.

Reply:
Each user can have private notes on all the chat reports they are part of. If I have access to reports 1 and 2, then I can have private notes on both report 1 and report 2. We also discussed bringing public notes, so in that case, they won't be tied to accounts but only to the reports.


> Ideally we shouldn't have to load a report just to get the privateNotesReportID for an account.

💯 agree with this. But it looks like we do that for a couple more things, no? like lastReadTime and stuff? Since there are no major concerns with how we are handling the data, I too will just stick with :donothing: for now.

Reply:
> I seriously can't recall why I went with this change, I have reverted back everything to get rid of double withOnyx usage.

Now I know why I made that change
https://github.com/Expensify/App/pull/28200#pullrequestreview-1659600378

Reply:
Let me just reiterate the problem statement so it's clear for you.

We have now added the notification preference option on the profile page. The challenge is that the profile page doesn't subscribe to any report by default. Instead, it depends on the user profile you've navigated to. It's not just about fetching the chat report between you and another user; the ProfilePage needs to actively subscribe to it. This subscription is crucial because when you update your notification preference, the component should re-render and display the updated notification preference from the report.

This was first reported here: https://github.com/Expensify/App/pull/28200#issuecomment-1736658690 and we did figure out a better solution for this here https://github.com/Expensify/App/pull/28200#discussion_r1338051983, which involved creating a separate component for the notification preference. This new component would subscribe to the DM chat report using Onyx, ensuring that when you update your notification preference, it automatically re-renders and displays the updated data.

we agreed that it might be overkill https://github.com/Expensify/App/pull/28200#issuecomment-1749311478 so we just so we opted to stick with a double Onyx call for now. A future solution is to create a dedicated notification preference page component, which can be reused throughout the entire application as needed.